sympy/polys/matrices/domainmatrix.py

```diff
@@ -1927,12 +1933,116 @@ def rref_den(self):
+
+            The sparse implementations are always used by default but if the
+            method is e.g. ``'GJ_dense'`` then the dense implementation will
+            be used. The returned matrix will always have the same format
+            (sparse or dense) as the input.
+
+        keep_domain : bool, optional
+            If True (the default), the domain of the returned matrix and
+            denominator are the same as the domain of the input matrix. If
+            False, the domain of the returned matrix might be changed to an
+            associated ring or field.
+
+        Returns
+        =======
+
+        (DomainMatrix, scalar, list)
+            Reduced-row echelon form, denominator and list of pivot indices.
+
+        Notes
+        =====
+
+        In some cases it is more efficient to change to a different domain when
+        computing the RREF. For example, if the input is a sparse matrix over
+        :ref:`ZZ` it might be more efficient to convert to :ref:`QQ` and
+        compute the RREF using division. If ``keep_domain`` is ``True``, then
+        in this case the result would be converted back to :ref:`ZZ` by
+        clearing denominators. This conversion is potentially expensive, so if
+        the caller does not need the result to be in :ref:`ZZ`, it is more
+        efficient to set ``keep_domain`` to ``False`` and let the result be in
+        :ref:`QQ`. Similarly, if the input is a matrix over :ref:`QQ` then if
+        ``keep_domain`` is ``False`` the result might be returned over
+        :ref:`QQ` or :ref:`ZZ` depending on which algorithm is used.
+
+        See Also
+        ========
+
+        rref
+            RREF without denominator for field domains.
+        sympy.polys.matrices.rref.dm_rref_den
+            The function that implements this.
+
+        """
+        return dm_rref_den(self, method=method, keep_domain=keep_domain)
+
+    def rref_gj_div(self):
+        """Compute RREF using Gauss-Jordan elimination with division.
+
+        The domain must be a field.
+
+        Returns
+        =======
+
+        (DomainMatrix, list)
+            reduced-row echelon form and list of pivots for the DomainMatrix
+
         See Also
         ========
 
-        convert_to, lu
         rref
+            Compute RREF choosing the most efficient algorithm (possibly uses
+            this algorithm).
+        rref_den
+            Compute RREF with denominator using the most efficient algorithm
+            (possibly uses this algorithm).
+        sympy.polys.matrices.dense.ddm_irref
+            The underlying algorithm in the dense case.
+        sympy.polys.matrices.sdm.sdm_irref
+            The underlying algorithm in the sparse case.
+        """
+        if not self.domain.is_Field:
+            raise DMNotAField('Not a field')
+        rref_ddm, pivots = self.rep.rref()
+        return self.from_rep(rref_ddm), tuple(pivots)
 
+    def rref_den_gj_ff(self):
```

Comment:
Also, I would like to not to add them, especially it is controlled by `method`s.
I'm not sure if you left it by intent, but it looks like some clutter.